plugins/outputs/timestream/sample.conf

```diff
@@ -47,6 +47,26 @@
   ## Specifies if the plugin should create the table, if the table does not exist.
   create_table_if_not_exists = true
 
+  ## Specifies if table should be created with composite partition key
+  ## that determines how your data will be stored and distributed.
+  ## create_table_composite_partition_key is an array of array with type string
+  ## Dimension type partition key requires 3 elements:
```

Comment:
I see no discussion about the difference between a dimension and a measure. From the link you provided I do not see it mentioned there as well. Can you please explain what the difference is?

I am not sold on this format or the user either. Having this strange array of array with different formats for two types of keys sounds error prone. Additionally it is weird to have a boolean field be a string when it is not anywhere else.

What do you think about having two options one for measurement type partition and one for dimension?

Reply:
Hi powersj@, thanks for the comments.

The reason we want to use an array of array is that even though Timestream only supports using one partition key at the moment, we plan to support using multiple partition keys in the future. From our definition of composite partition keys, we need to have an enforcement field and also dimension name field for partition key with type dimension. In addition, we want to preserve the order of the keys user inputs so we shouldn't separate the keys based on their types.

Let me know if that makes sense. Also addressed your other comments as suggested.

Reply:
@mhchengaws looking at the configuration option I think @powersj is right. I would really like to see something like

```toml
  create_table_composite_partition_keys = [
    { type = "Dimension", name = "you_name_it", enforce=true" },
    { type = "Measurement", name = "another_name" }
  ]
```

allowing to have defaults, e.g. for `enforce` **and** making it easier to work with those settings in the code. What do you think?